### PR TITLE
WIP: Add Reuters-21578 dataset

### PR DESCRIPTION
**Pull request by achyudhk**
I hope the commit messages are descriptive enough of the changes made. This PR is complementary to https://git.uwaterloo.ca/jimmylin/Castor-data/merge_requests/7#note_71256 in the data repository.

### CODE REVIEW

File: common/evaluators/reuters_evaluator.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        self.data_loader.init_epoch()
+        n_dev_correct = 0
+        total_loss = 0
+
+        for batch_idx, batch in enumerate(self.data_loader):
+            scores = self.model(batch.text)
+            # Using binary accuracy
+            for tensor1, tensor2 in zip(scores.round().long().data, batch.label.data):
+                print(tensor1, tensor2)
+                if np.array_equal(tensor1, tensor2):
```
Comment (Impavidity) [inline common/evaluators/reuters_evaluator.py]: So your evaluation criteria is that: give full credit when it is exact match otherwise, it get zero ?

File: common/evaluators/reuters_evaluator.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        self.model.eval()
+        self.data_loader.init_epoch()
+        n_dev_correct = 0
+        total_loss = 0
+
+        for batch_idx, batch in enumerate(self.data_loader):
+            scores = self.model(batch.text)
+            # Using binary accuracy
+            for tensor1, tensor2 in zip(scores.round().long().data, batch.label.data):
+                print(tensor1, tensor2)
```
Comment (Impavidity) [inline common/evaluators/reuters_evaluator.py]: remove this

File: common/evaluators/reuters_evaluator.py (hunk 1 above)
Comment (achyudhk) [reply common/evaluators/reuters_evaluator.py]: Yup. Since only one or two in 90 classes would be the positive label, I'm not sure if something like hamming score would work. The fraction of labels that are incorrectly predicted would still be too small in that case,

File: common/evaluators/reuters_evaluator.py (hunk 2 above)
Comment (achyudhk) [reply common/evaluators/reuters_evaluator.py]: My bad. I made the change, forgot to push it.
